py/selenium/webdriver/firefox/firefox_profile.py

```diff
@@ -330,6 +330,20 @@ def get_text(element):
                     rc.append(node.data)
             return ''.join(rc).strip()
 
+        def parse_manifest_json(content):
+            """Extracts the details from the contents of a WebExtensions `manifest.json` file."""
+            manifest = json.loads(content)
+            try:
+                id = manifest['applications']['gecko']['id']
+            except KeyError:
+                id = manifest['name'].replace(" ", "") + "@" + manifest['version']
```

Comment:
Thank you for picking this up! I've looked into it more and discovered that whilst web extensions don't mandate an id, it appears that when installing extensions from the profile [Firefox will verify that the name of the file/directory containing the extension matches the id of the add-on](https://searchfox.org/mozilla-central/rev/819cd31a93fd50b7167979607371878c4d6f18e8/toolkit/mozapps/extensions/internal/XPIDatabase.jsm#2420-2422). This means that we effectively do require an id to be set in the manifest, and that we use it when writing to the profile directory. It may work by coincidence when the id matches the name@version format used here. It's worth testing with the other clients, but if this is the case then I believe we should require the id and raise an exception when it's not provided to avoid harder to debug issues with Firefox launching without the addon installed. Sorry for my earlier comment suggesting that we needed id to be optional, which was based on [this documentation](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/applications).

Reply:
True! Reported as https://github.com/SeleniumHQ/selenium/issues/6471